app/validators/personal_key_validator.rb

```diff
@@ -0,0 +1,44 @@
+module PersonalKeyValidator
+  extend ActiveSupport::Concern
+
+  included do
+    include ActiveModel::Validations::Callbacks
+
+    before_validation :normalized_code
```

Comment:
Like I said over screenshare, I don't like this sneaky method with a side effect, I would prefer an explicit `normalize_personal_key(personal_key)` call in the initializers